check_source.py

```diff
@@ -150,12 +151,25 @@ def check_source_submission(self, source_project, source_package, source_revisio
             self.logger.info('checking if target package exists and has devel project')
             devel_project, devel_package = devel_project_get(self.apiurl, target_project, target_package)
             if devel_project:
-                if ((source_project != devel_project or source_package != devel_package) and
-                   not (source_project == target_project and source_package == target_package)):
-                    # Not from proper devel project/package and not self-submission.
-                    self.review_messages['declined'] = 'Expected submission from devel package %s/%s' % (
-                        devel_project, devel_package)
-                    return False
+                if (
+                        (source_project != devel_project or source_package != devel_package)
+                        and not (source_project == target_project and source_package == target_package)):
+                    # check if the devel project & package are using scmsync & match the allowed prj prefix
+                    # => waive the devel project source submission requirement
+                    meta = ET.fromstringlist(show_package_meta(self.apiurl, devel_project, devel_package))
+                    scm_sync = meta.find('scmsync')
+                    if (
+                            not scm_sync or
+                            (
+                                scm_sync and
+                                not scm_sync.text.startswith(f"https://src.opensuse.org/pool/{source_package}")
+                                and all(not source_project.startswith(allowed_src) for allowed_src in self.allowed_scm_submission_sources)
```

Comment:
isn't that supposed to be `any(` rather than `all(` ?

Reply:
No, the branch should only be taken if the source project name begins with something that is none of the allowed sources.